openupgrade_scripts/scripts/website_sale/15.0.1.1/post-migration.py

```diff
@@ -1,6 +1,40 @@
+import re
+
 from openupgradelib import openupgrade
 
 
+def replace_content_in_product_custom_text(env):
+    """Replace Terms and Conditions content in the new v15 template so as not to lose
+    content from previous versions if it has been customised."""
+    product_views = env["ir.ui.view"].search(
+        [("key", "=", "website_sale.product"), ("website_id", "!=", False)]
+    )
+    for view in product_views:
+        product_arch = view.arch_db
+        product_pattern = r'<p class="text-muted"(.*?)<\/p>'
+        product_matches = re.findall(product_pattern, product_arch, re.DOTALL)
+
+        if product_matches:
+            product_custom_text_view = env["ir.ui.view"].search(
+                [("key", "=", "website_sale.product_custom_text")]
```

Comment:
```suggestion
                [("key", "=", "website_sale.product_custom_text"), ("website_id", "=", False)]
```

and do this search only once at the beginning and don't perform the rest if it doesn't exist.